lienzo-core/src/main/java/com/ait/lienzo/client/core/shape/wires/ILineSpliceAcceptor.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ait.lienzo.client.core.shape.wires;
+
+import com.ait.lienzo.client.core.types.Point2D;
+import com.ait.lienzo.client.core.types.Point2DArray;
+
+public interface ILineSpliceAcceptor {
+
+    ILineSpliceAcceptor ALL = new DefaultLineSpliceAcceptor(true);
+
+    ILineSpliceAcceptor NONE = new DefaultLineSpliceAcceptor(false);
+
+    boolean allowSplice(WiresShape shape,
+                        Point2D candidateLocation,
+                        WiresConnector connector,
+                        WiresContainer parent);
+
+    boolean acceptSplice(WiresShape shape,
+                         Point2D candidateLocation,
+                         WiresConnector connector,
+                         Point2DArray firstHalfPoints,
```

Comment:
another quick Q - I see the arguments for the `acceptSplice` contain `firstHalfPoints` and `secondHalfPoints`, but is this really necessary? I mean:

- the points array can be obtained from the `connector` instance, which is alrady an argument
- this way, buy just adding an `integer` argument, which specificies the index where to split the above connector's points, we can extrapolate same info
- so given the `connector` and some `index` (int), the `firstHalfPoints` and `secondHalfPoints` can be extrapolated

Does this makes sense? Or is not so easy? :)

Again, this not blocks this PR, just brainstorming,